url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/alg/dense/linsol/lu/LinearSolverLuKJI.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2009-2014, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.alg.dense.linsol.lu;
+
+import org.ejml.alg.dense.decomposition.lu.LUDecompositionBase_D64;
+import org.ejml.data.DenseMatrix64F;
+import org.ejml.ops.SpecializedOps;
+
+
+/**
+ * To avoid cpu cache issues the order in which the arrays are traversed have been changed.
+ * There seems to be no performance benit relative to {@link LinearSolverLu} in this approach
+ * and b and x can't be the same instance, which means it has slightly less functionality.
+ *
+ * @author Peter Abeles
+ */
+public class LinearSolverLuKJI extends LinearSolverLuBase {
+
+    private double []dataLU;
+    private int[] pivot;
+
+    public LinearSolverLuKJI( LUDecompositionBase_D64 decomp ) {
+        super(decomp);
+
+    }
+
+    @Override
+    public boolean setA(DenseMatrix64F A) {
+        boolean ret = super.setA(A);
+
+        pivot = decomp.getPivot();
+        dataLU = decomp.getLU().data;
+
+        return ret;
+    }
+
+    /**
+     * An other implementation of solve() that processes the matrices in a different order.
+     * It seems to have the same runtime performance as {@link #solve} and is more complicated.
+     * It is being kept around to avoid future replication of work.
+     *
+     * @param b A matrix that is n by m.  Not modified.
+     * @param x An n by m matrix where the solution is writen to.  Modified.
+     */
+    @Override
+    public void solve(DenseMatrix64F b, DenseMatrix64F x) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.